test_conformance/basic/test_int2fp.cpp

```diff
@@ -0,0 +1,327 @@
+//
+// Copyright (c) 2023 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "CL/cl_half.h"
+#include "harness/compat.h"
+#include "harness/errorHelpers.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#include <algorithm>
+#include <map>
+#include <vector>
+
+#include "procs.h"
+#include "utils.h"
+
+namespace {
+const char *int2float_kernel_code = R"(
+%s
+__kernel void test_X2Y(__global TYPE_X *src, __global TYPE_Y *dst)
+{
+    int  tid = get_global_id(0);
+
+    dst[tid] = (TYPE_Y)src[tid];
+
+})";
+
+// half in range -32768..32768, non-zero exp and mantissa
+cl_half get_random_half(MTdata d)
+{
+    const unsigned m_size = 0x3fe;
+    const unsigned e_size = 0x1d;
+    const unsigned s_size = 0x1;
+    unsigned ind = genrand_int32(d) % (s_size * e_size * m_size);
+    return static_cast<cl_half>((((ind / (e_size * m_size)) << 15)
+                                 | (((ind / m_size) % e_size + 1) << 10)
+                                 | (ind % m_size + 1)));
+}
```

Comment:
I'm not quite following this code, first of all why it's needed it'd expect zero mantissa numbers to be some of the easier values to convert as there's no fractional component.

Secondly, once it's documented why we need this function itself needs more comments. There's a lot of expressions in here, and i'd prefer them to be in their own statements with comments for this code to be maintainable.

Reply:
This code has been replaced with generator based on `get_random_float` in specific range:

```
-MAKE_HEX_FLOAT(0x1.ffcp+14, 1.9990234375f, 14),
 MAKE_HEX_FLOAT(0x1.ffcp+14, 1.9990234375f, 14)
```

Previous procedure turns out to be a temporary solution for test failing on my local device at the border values.